app/src/main/java/com/duckduckgo/widget/SearchAndFavoritesWidget.kt

```diff
@@ -97,45 +108,56 @@ class SearchAndFavoritesWidget : AppWidgetProvider() {
         newOptions: Bundle
     ) {
         Timber.i("SearchAndFavoritesWidget - onAppWidgetOptionsChanged")
-        updateWidget(context, appWidgetManager, appWidgetId, newOptions)
+        appCoroutineScope.launch {
+            updateWidget(context, appWidgetManager, appWidgetId, newOptions)
+        }
         super.onAppWidgetOptionsChanged(context, appWidgetManager, appWidgetId, newOptions)
     }
 
     override fun onDeleted(
         context: Context,
         appWidgetIds: IntArray
     ) {
-        appWidgetIds.forEach {
-            widgetPrefs.removeWidgetSettings(it)
+        appCoroutineScope.launch(Dispatchers.IO) {
```

Comment:
Should probably provide `DispatcherProvider` rather than using this directly

Reply:
We don't have that currently DI-injectable 🤔. is it worth manually creating the `DefaultDispatcherProvider()` here even when there is no other thing overriding it? e.g., `private val dispatchers = DefaultDispatcherProvider()`

Happy to do it if that seems like an improvement.

Reply:
We should have that though, i.e. `BookmarksViewModel` has it, also `BrokenSiteSubmitter` and quite a few others. It is provided in `CoroutinesModule`

Reply:
Sweet, hadn't realised it was defined there. _sold!_